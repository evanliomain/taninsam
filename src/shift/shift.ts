/**
 * @module array=>array
 */
/**
 * Removes the first element from an array.
 * @return the function to apply on the array
 * @example
 * ```
 * shift<number>()([1, 2, 3, 4, 5]) // [1, 2, 3, 4]
 * ```
 * @example Using the chain
 * ```
 * chain([1, 2, 3, 4, 5])
 *   .chain(shift<number>())
 *   .value() // [1, 2, 3, 4]
 * ```
 */
export function shift<T>(): (array: ReadonlyArray<T>) => ReadonlyArray<T> {
  return ([_, ...tail]: ReadonlyArray<T>) => tail;
}
