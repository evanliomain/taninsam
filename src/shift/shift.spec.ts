import { shift } from './shift';

describe('shift function', () => {
  const input = [1, 2, 3, 4, 5];

  test('is a pure function', () => {
    shift()(input);
    expect(input).toMatchSnapshot();
  });

  test('[1, 2, 3, 4, 5] |> shift === [2, 3, 4, 5]', () => {
    expect(shift()(input)).toMatchSnapshot();
  });

  test('[] |> shift === []', () => {
    expect(shift()([])).toMatchSnapshot();
  });
});
