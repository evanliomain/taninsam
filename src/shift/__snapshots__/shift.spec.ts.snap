// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`shift function > [] |> shift === [] 1`] = `[]`;

exports[`shift function > [1, 2, 3, 4, 5] |> shift === [2, 3, 4, 5] 1`] = `
[
  2,
  3,
  4,
  5,
]
`;

exports[`shift function > is a pure function 1`] = `
[
  1,
  2,
  3,
  4,
  5,
]
`;
